### PR TITLE
Remove support for prometheus 1.x

### LINKED ISSUE #2813: Dropping support for prometheus 1.x
Prometheus 2.0 was released almost 2 years ago, maybe it is time to move on and remove support for 1.x from the operator code base. 

Lots of the advanced features that this operator is bringing are not even available on prometheus 1.x. Additionally we test only upgrade path from 1.x to 2.x and not if features run on 1.x, which begs to question how much of operator features are supported in 1.4 (oldest version mentioned in [Documentation/compatibility.md](https://github.com/coreos/prometheus-operator/blob/master/Documentation/compatibility.md)?

cc: @brancz, @metalmatze, @s-urbaniak, @squat, @LiliC, @bwplotka

### PR DESCRIPTION
**Pull request by paulfantom**
Closes #2813 

**Release Note Template (will be copied)**

```release-note:change
Drop support for prometheus 1.x
```

### CODE REVIEW

**Comment by brancz** [comment]
We can do this, but after we are done with all the other in progress work. Let’s finish up the features for this release first, then we can consider this.

**Comment by lilic** [comment]
Yes please! 👍

File: pkg/prometheus/statefulset_test.go [hunk 1]
```diff
@@ -333,33 +333,6 @@ func TestStatefulSetVolumeInitial(t *testing.T) {
 
 }
 
-func TestMemoryRequestNotAdjustedWhenLimitLarger2Gi(t *testing.T) {
```
Comment (paulfantom) [inline pkg/prometheus/statefulset_test.go:336]: This was testing a feature existing only in prometheus 1.x path.

File: pkg/prometheus/statefulset_test.go [hunk 2]
```diff
@@ -391,33 +364,6 @@ func TestAdditionalConfigMap(t *testing.T) {
 	}
 }
 
-func TestMemoryRequestAdjustedWhenOnlyLimitGiven(t *testing.T) {
```
Comment (paulfantom) [inline pkg/prometheus/statefulset_test.go:394]: This was testing a feature existing only in prometheus 1.x path.

**Review by simonpasquier — COMMENTED** [review]
lgtm
:ship:

**Review by lilic — APPROVED** [review]
Thank you! lgtm

Note to the release person: Lets add a bigger note in the CHANGELOG, as we are deprecating something, in case someone is still using 1.x :)

**Pull request merged**